Got it. This is the build configuration
- Host: Arch linux (or in this case, any linux with default UID as 1000 can trigger the bug)
- Command: `make docker-build` in `docker` folder
### if USE_LOCAL_CREDENTIALS is NOT set
- the Makefile build docker with `build` and `ubuntu` user (which return no error)
- based on this command in `make docker-run`, nothing will be mount, result in empty container
```dockerfile
	@$(eval DOCKER_RUNARGS += $(if $(USE_LOCAL_CREDENTIALS),--volume=$(DOCKER_WORKMNT),))
	@$(eval DOCKER_RUNARGS += $(if $(USE_LOCAL_CREDENTIALS),--network=host,))
	@$(eval DOCKER_RUNARGS += $(if $(USE_LOCAL_CREDENTIALS),--volume=/gro/cad:/cad,))
	@$(eval DOCKER_RUNARGS += $(if $(USE_LOCAL_CREDENTIALS),--volume=/tools/Xilinx:/cad/Xilinx,))
```
- it is a slient fail to mount, so to warn, I add this line to Makefile
```makefile
# User Guidance
ifeq ($(USE_LOCAL_CREDENTIALS),)
$(warning [WARNING] Building/Running WITHOUT local credentials.)
$(warning [WARNING] For local development, set USE_LOCAL_CREDENTIALS otherwise the content WON'T be mounted)
endif
```
### if USER_LOCAL_CREDENTIALS is SET
- the initial build will return this error, happen because the host have the same default UID 1000
```
#12 [ 7/19] RUN useradd --no-log-init --uid 1000 --gid 1000 --home-dir /work test
#12 0.069 useradd: UID 1000 is not unique
#12 ERROR: process "/bin/sh -c useradd --no-log-init --uid $USER_ID --gid $GROUP_ID --home-dir $WORKDIR $USER_NAME" did not complete successfully: exit code: 4
------
 > [ 7/19] RUN useradd --no-log-init --uid 1000 --gid 1000 --home-dir /work test:
0.069 useradd: UID 1000 is not unique
------
Dockerfile.ubuntu24.04:40
--------------------
  38 |     
  39 |     # Use --no-log-init to deal with large userids creating giant log files
  40 | >>> RUN useradd --no-log-init --uid $USER_ID --gid $GROUP_ID --home-dir $WORKDIR $USER_NAME
  41 |     
  42 |     # Add the user to other groups
--------------------
ERROR: failed to build: failed to solve: process "/bin/sh -c useradd --no-log-init --uid $USER_ID --gid $GROUP_ID --home-dir $WORKDIR $USER_NAME" did not complete successfully: exit code: 4
make: *** [Makefile:46: build-ubuntu24.04] Error 1
```
- to address, I add this delete ubuntu user (centos didn't have this problem, so no change in centos)
```dockerfile
# Only delete the 'ubuntu' user if its UID (1000) matches the one we need to create
RUN if [ "$USER_ID" = "1000" ]; then \
		touch /var/mail/ubuntu && chown ubuntu /var/mail/ubuntu && userdel -r ubuntu || true; \
	fi
```
- after the change, the next problem encoutered is the group assignment because there's exists group that already been created in container, yet the dockerfile try copy from host (which stop the copy, then wrack the naming)
```
------
 > [ 9/20] RUN for gid in 1000 90 964 968 985 986 994 998; do         echo "group$gid";         groupadd -g $gid "group$gid" || true;         usermod -aG "group$gid" test;     done:
0.097 usermod: group 'group1000' does not exist
0.097 group90
0.127 group964
0.151 group968
0.176 group985
0.202 group986
0.229 group994
0.255 group998
0.257 groupadd: GID '998' already exists
0.258 usermod: group 'group998' does not exist
------
Dockerfile.ubuntu24.04:48
--------------------
  47 |     # Add the user to other groups
  48 | >>> RUN for gid in ${OTHER_GROUPS}; do \
  49 | >>>         echo "group$gid"; \
  50 | >>>         groupadd -g $gid "group$gid" || true; \
  51 | >>>         usermod -aG "group$gid" $USER_NAME; \
  52 | >>>     done
  53 |     
--------------------
ERROR: failed to build: failed to solve: process "/bin/sh -c for gid in ${OTHER_GROUPS}; do         echo \"group$gid\";         groupadd -g $gid \"group$gid\" || true;         usermod -aG \"group$gid\" $USER_NAME;     done" did not complete successfully: exit code: 6
make: *** [Makefile:46: build-ubuntu24.04] Error 1
```
- to address, I fix the loop to try get the name of the group first, if not exists then create new one

--- a/docker/Dockerfile.centos7
+++ b/docker/Dockerfile.centos7
@@ -53,15 +53,8 @@ RUN useradd --no-log-init --uid $USER_ID --gid $GROUP_ID --home-dir $WORKDIR $US
 
 # Add the user to other groups
 RUN for gid in ${OTHER_GROUPS}; do \
-		# Check if a group with this GID already exists in the container
-		# Then add user accordingly
-		EXISTING_NAME=$(getent group $gid | cut -d: -f1); \
-		if [ -n "$EXISTING_NAME" ]; then \
-			usermod -aG "$EXISTING_NAME" $USER_NAME; \
-		else \
-			groupadd -g $gid "group$gid" && \
-			usermod -aG "group$gid" $USER_NAME; \
-		fi; \
+		groupadd -g $gid "group$gid" || true; \
+		usermod -aG "$(getent group $gid)" "$USER_NAME"; \
     done
 
 # Create directory for extra installation

--- a/docker/Dockerfile.ubuntu24.04
+++ b/docker/Dockerfile.ubuntu24.04
@@ -33,10 +33,8 @@ ARG OTHER_GROUPS
 ARG WORKDIR=/work
 ENV DOCKERFILE_PLATFORM="ubuntu24.04"
 
-# Only delete the 'ubuntu' user if its UID (1000) matches the one we need to create
-RUN if [ "$USER_ID" = "1000" ]; then \
-		touch /var/mail/ubuntu && chown ubuntu /var/mail/ubuntu && userdel -r ubuntu || true; \
-	fi
+# Delete default ubuntu user
+RUN userdel -r ubuntu || true;
 
 # Only create the group if it doesn't exist
 RUN getent group $GROUP_ID || groupadd --gid $GROUP_ID $GROUP_NAME
@@ -46,15 +44,8 @@ RUN useradd --no-log-init --uid $USER_ID --gid $GROUP_ID --home-dir $WORKDIR $US
 
 # Add the user to other groups
 RUN for gid in ${OTHER_GROUPS}; do \
-		# Check if a group with this GID already exists in the container
-		# Then add user accordingly
-		EXISTING_NAME=$(getent group $gid | cut -d: -f1); \
-		if [ -n "$EXISTING_NAME" ]; then \
-			usermod -aG "$EXISTING_NAME" $USER_NAME; \
-		else \
-			groupadd -g $gid "group$gid" && \
-			usermod -aG "group$gid" $USER_NAME; \
-		fi; \
+		groupadd -g $gid "group$gid" || true; \
+		usermod -aG "$(getent group $gid)" "$USER_NAME"; \
     done
 
 # Create directory for extra installation

